### ISSUE TITLE
Consider publishing this in the Windows Store

### ISSUE THREAD

**Issue by surban** [uutils/coreutils#2758]
Although it is not widely known, the Windows Store allows installation of command line binaries into a directory that is in the user's PATH.  Having this collection of tools available to Windows users with a straightforward method for installation and upgrades would be a huge benefit.

**Comment by sylvestre** [comment]
Interesting idea! 
Is there "low level" tools on the Windows store already?
And pure rust software ? (besides Firefox)

**Comment by surban** [comment]
Powershell and Python are available in the store, see https://www.microsoft.com/store/productId/9MZ1SNWT0N5D and https://www.microsoft.com/store/productId/9PJPW5LDXLZ5.

`ls` is also available on the store, but without a trusted publisher and link to a repository I wouldn't trust it.

I don't know about Rust software, but it shouldn't matter, since Rust is statically linked and you can use the full Win32 API from the store.

**Comment by sylvestre** [comment]
Cool, thanks for investigating! :)

**Comment by YDX-2147483647** [comment]
I research it today, and I am afraid that Microsoft Store is not suitable for things like coreutils…

[Python Doc](https://docs.python.org/3/using/windows.html#redirection-of-local-data-registry-and-temporary-paths):

> Because of restrictions on Microsoft Store apps, Python scripts may **not have full write access to shared locations** such as `TEMP` and the registry. Instead, it will write to a private copy. If your scripts must modify the shared locations, you will need to install the full installer.
>
> …
>
> When reading files, Windows will return the file from the private folder, or if that does not exist, the real Windows directory. For example reading `C:\Windows\System32` returns the contents of `C:\Windows\System32` plus the contents of `C:\Program Files\WindowsApps\package_name\VFS\SystemX86`.

**Comment by beedell-roke** [comment]
> https://github.com/uutils/coreutils/issues/2758#issuecomment-1812587269

@YDX-2147483647, that's not a showstopper — just publishing the coreutils as an unsandboxed `.MSIX` file to [`microsoft/winget-pkgs`](https://github.com/microsoft/winget-pkgs) provides exactly the same benefits. Some WinGet packages are even accessible via the Microsoft Store GUI.

**Comment by oech3** [comment]
Does MS Store and winget allow installing hardlinking binary? (Symlink is not allowed by default)

**Comment by beedell-roke** [comment]
> Does MS Store and winget allow installing hardlinking binary?

@oech3, I don't know about whether the `msstore` remote does, but `winget-pkgs` does; it's significantly more like `brew`'s manifest-for-external-software, than `dnf5` or `apt`'s curated repositories, which may be the origin of your confusion. To demonstrate, `coreutils` already exists at the `winget` remote, per [`microsoft/winget-pkgs/issues/126833#event-11042181875`](https://github.com/microsoft/winget-pkgs/issues/126833#event-11042181875). However, these locations are not equivalent, so this is tangential.

**Comment by oech3** [comment]
At next release, .zip will contain individual binaries. No longer needed to care about support of hardlinkings at winget, Windows Store.

@RokeJulianLockhart Are you interested in submitting latest-commit to winget?
https://github.com/uutils/coreutils/releases/download/latest-commit/coreutils-x86_64-pc-windows-msvc.zip

**Comment by beedell-roke** [comment]
@oech3, shouldn't `winget-pkgs` automatically catch any new `/releases`, via `/latest`? If `latest-commit` should permanently be the new target, I suppose that we can submit a package issue, although unstable versions are generally best packaged as a new package, if at all.

**Comment by oech3** [comment]
~What `winget` command to install it?~
needed to add `latest-commit` to https://github.com/microsoft/winget-pkgs/tree/master/manifests/u/uutils/coreutils ?

**Comment by beedell-roke** [comment]
@oech3, I don't understand what you're asking. The sentence is phrased as if it's a statement, yet punctuated like a question. [`microsoft/winget-pkgs/issues/new/choose`](https://github.com/microsoft/winget-pkgs/issues/new/choose) may be useful for you.

**Comment by oech3** [comment]
Hmm... If I understand winget, it does not accept a package without frozen checksum. So we can't submit latest-commit to there...

**Comment by beedell-roke** [comment]
@oech3, indeed, per [`microsoft/winget-pkgs/issues/163098`](https://github.com/microsoft/winget-pkgs/issues/163098#issue-2409248645). However, this ticket is about the `msstore` remote. Not `winget`.

**Comment by oech3** [comment]
Does MSStore accept package without checksum e.g. nightly bins. If so, it does not require too many update.
